Frameworks/Foundation/_NSUndoManagerStack.mm

```diff
@@ -0,0 +1,47 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <Starboard.h>
+#import "_NSUndoManagerInternal.h"
+#include <list>
+
+@implementation _NSUndoManagerStack {
+    std::list<StrongId<id<_NSUndoable>>> _stack;
+}
+
+- (void)pop {
+    if ([self count] > 0) {
+        _stack.pop_front();
+    }
+}
+
+- (id<_NSUndoable>)peek {
+    if ([self count] > 0) {
+        return _stack.front();
+    }
+    return nil;
+}
+
+- (void)push:(id<_NSUndoable>)undoObject {
+    _stack.push_front(undoObject);
+    [undoObject release];
```

Comment:
My advice to release this only applied if you were pushing it onto the std::list in the same place you were allocating it.

If you think about it, this means that `push:` is now a consumer for a +1 reference. That's not what something named push should do.

Why not just have the `std::list`s right on `NSUndoManager`? It seems like we could avoid the dynamic dispatch here.